chemprop/cli/utils/parsing.py

```diff
@@ -199,28 +200,83 @@ def make_datapoints(
     else:
         N = len(smiss[0])
 
+    if len(smiss) > 0:
+        molss = [[make_mol(smi, keep_h, add_h) for smi in smis] for smis in smiss]
+    if len(rxnss) > 0:
+        rctss = [
+            [
+                make_mol(f"{rct_smi}.{agt_smi}" if agt_smi else rct_smi, keep_h, add_h)
+                for rct_smi, agt_smi, _ in (rxn.split(">") for rxn in rxns)
+            ]
+            for rxns in rxnss
+        ]
+        pdtss = [
+            [make_mol(pdt_smi, keep_h, add_h) for _, _, pdt_smi in (rxn.split(">") for rxn in rxns)]
+            for rxns in rxnss
+        ]
+
     weights = np.ones(N, dtype=np.single) if weights is None else weights
     gt_mask = [None] * N if gt_mask is None else gt_mask
     lt_mask = [None] * N if lt_mask is None else lt_mask
 
     n_mols = len(smiss) if smiss else 0
-    X_d = [None] * N if X_d is None else X_d
     V_fss = [[None] * N] * n_mols if V_fss is None else V_fss
     E_fss = [[None] * N] * n_mols if E_fss is None else E_fss
     V_dss = [[None] * N] * n_mols if V_dss is None else V_dss
 
+    if X_d is None and molecule_featurizers is None:
+        X_d = [None] * N
+    else:
+        if molecule_featurizers is None:
+            pass
+        else:
```

Comment:
I think this can be modified to
```
if X_d is None and molecule_featurizers is None:
    X_d = [None] * N
elif molecule_featurizers is None:
    pass
elif molecule_featurizers is not None:
    ...
    if X_d is None:
        X_d = mol_descriptors
    else:
        X_d = np.hstack([X_d, mol_descriptors])
```

Reply:
That looks like a good change. I added that but used `else:` instead of `elif molecule_featurizer is not None:`. I think those are equivalent.